### PR TITLE
Add support for MassTransit Request and Response mechanisms to the library

### PR DESCRIPTION
**Pull request by xtremekforever**
This works with a C# example app, but the demo included does request/response between 2 tasks.

### CODE REVIEW

File: Sources/MassTransit/MassTransitUtils.swift [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import AsyncAlgorithms
+
+func randomString(length: Int) -> String {
+    let letters = "abcdefghijklmnopqrstuvwxyz0123456789"
+    var randomString = ""
+    for _ in 0..<length {
+        let randomIndex = Int.random(in: 0..<letters.count)
+        let letter = letters[letters.index(letters.startIndex, offsetBy: randomIndex)]
```
Comment (orobio) [inline Sources/MassTransit/MassTransitUtils.swift]: `letters.randomElement()!` works as well

File: Sources/MassTransit/MassTransitUtils.swift [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+func getModuleName<T>(_ module: T) -> String {
+    return String(String(reflecting: T.self).prefix { $0 != "." }).replacingOccurrences(of: "_", with: "")
+}
+
+func gracefulCancellableDelay(timeout: Duration) async throws {
+    for await _ in AsyncTimerSequence(interval: timeout, clock: .continuous).cancelOnGracefulShutdown() {
+        break
+    }
+}
```
Comment (orobio) [inline Sources/MassTransit/MassTransitUtils.swift]: The function is declared to be throwing, but it doesn't. I wonder if it should, though. I would expect the same semantics as Task.sleep: throwing on interruption.

**Review by orobio — COMMENTED** [review]
I don't know much about the Mass Transit details, but in general this looks fine to me. I added 2 minor notes.

File: Sources/MassTransit/MassTransitUtils.swift (hunk 2 above)
Comment (xtremekforever) [reply Sources/MassTransit/MassTransitUtils.swift]: Well, the AsyncTimerSequence doesn't throw, so I guess the top method doesn't need to throw either!

File: Sources/MassTransit/MassTransitUtils.swift (hunk 1 above)
Comment (xtremekforever) [reply Sources/MassTransit/MassTransitUtils.swift]: It does! Nice, thanks!

File: Sources/MassTransit/MassTransitUtils.swift (hunk 2 above)
Comment (orobio) [reply Sources/MassTransit/MassTransitUtils.swift]: I have not really looked at where this is used, but in general the problem I see with that is that the caller cannot know whether the specified time has passed or the function returned early.

**Pull request merged**